homework-selects/sql_questions.sql

```diff
@@ -3,56 +3,419 @@
+            SELECT LOCATION_ID
+            FROM LOCATIONS
+            WHERE COUNTRY_ID = 'US'
+        )
+    )
+);
+
 --37. the names of all employees whose salary is greater than 50% of their department’s total salary bill.
+SELECT A.FIRST_NAME || ' ' || A.LAST_NAME AS "FULL NAME"
+FROM EMPLOYEES A
+WHERE SALARY > (
+    SELECT (SUM(SALARY))*.5
+    FROM EMPLOYEES B
+    WHERE A.DEPARTMENT_ID = B.DEPARTMENT_ID
+);
+
 --38. the employee id, name ( first name and last name ), salary, department name and city for all
 --the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.
+SELECT E.EMPLOYEE_ID, E.FIRST_NAME || ' ' || E.LAST_NAME AS "FULL NAME", E.SALARY, L.CITY
+FROM EMPLOYEES E
+         LEFT JOIN DEPARTMENTS D on E.DEPARTMENT_ID = D.DEPARTMENT_ID
+         LEFT JOIN LOCATIONS L on D.LOCATION_ID = L.LOCATION_ID
+WHERE E.SALARY = (SELECT MAX(SALARY)
+                  FROM EMPLOYEES
+                  WHERE HIRE_DATE
+                            BETWEEN '01-jan-2002' AND '31-dec-2003');
+
 --39. the first and last name, salary, and department ID for all those employees who earn more than the average salary and arrange the list in descending order on salary.
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY > (
+    SELECT AVG(SALARY)
+    FROM EMPLOYEES
+)
+ORDER BY SALARY DESC;
+
 --40. the first and last name, salary, and department ID for those employees who earn more than the maximum salary of a department which ID is 40.
+SELECT FIRST_NAME,LAST_NAME,DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY IN(
+    SELECT SALARY
+    FROM EMPLOYEES
+    WHERE DEPARTMENT_ID = 40
+);
+
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+SELECT DEPARTMENT_NAME, DEPARTMENT_ID
+FROM DEPARTMENTS
+WHERE LOCATION_ID = (
+    SELECT LOCATION_ID
+    FROM DEPARTMENTS
+    WHERE DEPARTMENT_ID = 30
+);
+
 --42. the first and last name, salary, and department ID for all those employees who work in that department where the employee works who hold the ID 201.
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE DEPARTMENT_ID = (
+    SELECT DEPARTMENT_ID
+    FROM EMPLOYEES
+    WHERE EMPLOYEE_ID = 201
+);
+
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY IN(
+    SELECT SALARY
+    FROM EMPLOYEES
+    WHERE DEPARTMENT_ID =40
+);
+
 --44. the first and last name, salary, and department ID for those employees who earn more than the minimum salary of a department which ID is 40.
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY > ANY(
+    SELECT SALARY
```

Comment:
Where do you check what is the minimum salary?

Reply:
MIN ( SALARY)